modules/heat_conduction/src/constraints/ModularGapConductanceConstraint.C

```diff
@@ -175,13 +175,22 @@ ModularGapConductanceConstraint::setGapGeometryParameters(
   {
     if (coord_sys == Moose::COORD_XYZ)
     {
-      if (!params.isParamValid("cylinder_axis_point_1") ||
-          !params.isParamValid("cylinder_axis_point_2"))
+      if (params.isParamValid("cylinder_axis_point_1") &&
+          params.isParamValid("cylinder_axis_point_2"))
+      {
+        _p1 = params.get<RealVectorValue>("cylinder_axis_point_1");
+        _p2 = params.get<RealVectorValue>("cylinder_axis_point_2");
+      }
+      else if (_mesh.dimension() == 3)
         paramError("gap_geometry_type",
-                   "For 'gap_geometry_type = CYLINDER' to be used with a Cartesian model, "
+                   "For 'gap_geometry_type = CYLINDER' to be used with a Cartesian model in 3D, "
                    "'cylinder_axis_point_1' and 'cylinder_axis_point_2' must be specified.");
-      p1 = params.get<RealVectorValue>("cylinder_axis_point_1");
-      p2 = params.get<RealVectorValue>("cylinder_axis_point_2");
+      else
+      {
+        deduceGeometryParameters();
+        mooseInfoRepeated(
+            "ModularGapConductanceConstraint '", name(), "' deduced cylinder axis as ", _p1, _p2);
```

Comment:
Suggestion: Say explicitly that possible user input is disregarded (I don't think there is any additional check elsewhere)

Reply:
If only one point for the cylinder is specified? I should make that an error. Otherwise no user input should be disregarded.

Reply:
Yes, you are right. I had read it wrong.